projects/Mallard/src/download-edition/clear-issues.ts

```diff
@@ -17,70 +10,39 @@ import {
     issuesToDelete,
 } from 'src/helpers/files'
 
-// for cleaning up temporary files when the user hits 'delete all downlods'
-// NOTE: these hard coded names may change when rn-fetch-blob is updated
-const TEMP_FILE_LOCATIONS = [
-    `${RNFetchBlob.fs.dirs.DocumentDir}/`,
-    `${RNFetchBlob.fs.dirs.DocumentDir}/RNFetchBlob_tmp/`,
-]
-const RN_FETCH_TEMP_PREFIX = 'RNFetchBlobTmp'
-
-const handleDeleteFailure = async (error: Error, dir: string) => {
-    await pushTracking(
-        'tempFileRemoveError',
-        JSON.stringify(error),
-        Feature.CLEAR_ISSUES,
-    )
-    console.log(
-        `Error cleaning up temp issue files in directory ${dir}: `,
-        error,
-    )
-    errorService.captureException(error)
-}
-
-const removeTempFiles = () => {
-    const removeOrphanedTempFiles = async (dir: string) => {
-        try {
-            const isDir = await RNFetchBlob.fs.isDir(dir)
-            if (isDir) {
-                RNFetchBlob.fs
-                    .ls(dir)
-                    .then(files => {
-                        files
-                            .filter(f => f.startsWith(RN_FETCH_TEMP_PREFIX))
-                            .map(f => dir + f)
-                            .map(RNFetchBlob.fs.unlink)
-                    })
-                    .catch(error => handleDeleteFailure(error, dir))
-            }
-        } catch (error) {
-            handleDeleteFailure(error, dir)
-        }
+const clearDownloadsDirectory = async () => {
+    try {
+        const files = await RNFS.readDir(FSPaths.downloadRoot)
+        files.map(
+            async (file: RNFS.ReadDirItem) => await RNFS.unlink(file.path),
```

Comment:
This can throw an error and which cause the app to show a big red error message (noticed on android)

I have fixed this in this [PR](https://github.com/guardian/editions/pull/1268/files)

Reply:
Also `clearDownloadsDirectory` get called on App start which means `clearDownloadsDirectory` will try to delete download folder even when a background download is in progress. It would be good to check that `clearDownloadsDirectory` can't delete the download folder in that scenario otherwise we can endup losing downloaded data.

Reply:
In the first comment, the PR is referencing the deletion of temporary files, which this library doesn't use.

The function you have highlighted gets all the files in the download folder and deletes them. Not the download folder itself. I hit this problem when originally doing this PR.

Reply:
Ah ok, then it is okay. I thought this library also can suffer from concurrent `edit` access with temp files